### PR TITLE
First draft of 3.1.1 blogpost

### PR DESCRIPTION
**Pull request by Kordyjan**
Blogpost about 3.1.1 and about the first publicly available version of the compiler with `-Yscala-release` flag.

### CODE REVIEW

File: blog/_posts/2022-01-17-scala-3.1.1-released.md
Comment (prolativ) [inline blog/_posts/2022-01-17-scala-3.1.1-released.md:14]: Maybe add links to PRs?

File: blog/_posts/2022-01-17-scala-3.1.1-released.md
Comment (bishabosha) [inline blog/_posts/2022-01-17-scala-3.1.1-released.md:15]:
```suggestion
- Mirrors can be correctly summoned for hierarchical sum types compiled by Scala 3.0.x.
```

File: blog/_posts/2022-01-17-scala-3.1.1-released.md
Comment (prolativ) [inline blog/_posts/2022-01-17-scala-3.1.1-released.md:28]: I think we should use `3.1.2-RC1` here instead of just `3.1.2` for consistency.

Comment (prolativ) [inline blog/_posts/2022-01-17-scala-3.1.1-released.md:28]: The syntax for passing the option from the command line is `-Yscala-release 3.0` or `-Yscala-release:3.0`.  This won't work with `=` as far as I know.

Comment (prolativ) [inline blog/_posts/2022-01-17-scala-3.1.1-released.md:28]: It's lower case `@since`

File: blog/_posts/2022-01-17-scala-3.1.1-released.md
Comment (prolativ) [inline blog/_posts/2022-01-17-scala-3.1.1-released.md:30]: `@since`

File: blog/_posts/2022-01-17-scala-3.1.1-released.md
Comment (SethTisue) [reply blog/_posts/2022-01-17-scala-3.1.1-released.md:14]: Agree this needs further explanation, perhaps just in the form of a link; I cannot make sense of the description

File: blog/_posts/2022-01-17-scala-3.1.1-released.md
Comment (SethTisue) [inline blog/_posts/2022-01-17-scala-3.1.1-released.md:8]: I think you could omit the material about timing. It's normal for releases to slip a bit, especially around the holidays. People are primarily interested in the content of the release; few care whether it's a bit late.

If you feel you must account for the lateness, consider doing so much nearer end of the announcement, rather than the beginning. Leading with this sets a defensive, things-aren't-going-well tone. But things actually are going well (the delay is unimportant), so there's no need to set that tone.

File: blog/_posts/2022-01-17-scala-3.1.1-released.md
Comment (SethTisue) [inline blog/_posts/2022-01-17-scala-3.1.1-released.md:12]: Any mention of Scala 3.1.1-RC2 is supplementary material to omit or move to later, not important information to lead with.

"Scala 3.1.1 is the first patch version after the 3.1 minor release." doesn't add any meaning as far as I can see, I think you could just omit it. Or maybe condense to "Scala 3.1.1 is a patch release focused on bugfixes." I don't think "improvements in overall correctness and stability" adds any meaning and can be omitted.

File: blog/_posts/2022-01-17-scala-3.1.1-released.md
Comment (SethTisue) [inline blog/_posts/2022-01-17-scala-3.1.1-released.md:14]: This is confusing — I genuinely don't know what this means.

When you say "conversions", you mean "implicit conversions", or you mean something else?

Is `<:<` a "Scala 2 conversion"? (If so, why? Scala 3 has `<:<` too.)

File: blog/_posts/2022-01-17-scala-3.1.1-released.md
Comment (SethTisue) [inline blog/_posts/2022-01-17-scala-3.1.1-released.md:27]: too minor to merit release-noting, IMO

File: blog/_posts/2022-01-17-scala-3.1.1-released.md
Comment (SethTisue) [inline blog/_posts/2022-01-17-scala-3.1.1-released.md:25]: If this is important enough to release note, I think the wording could do more to convey what the actual importance of this is.

File: blog/_posts/2022-01-17-scala-3.1.1-released.md
Comment (SethTisue) [inline blog/_posts/2022-01-17-scala-3.1.1-released.md:28]: is it "new", or is it merely "improved"?

is there a link with more information?

File: blog/_posts/2022-01-17-scala-3.1.1-released.md
Comment (SethTisue) [inline blog/_posts/2022-01-17-scala-3.1.1-released.md:34]: Link to somewhere I can read about this?

File: blog/_posts/2022-01-17-scala-3.1.1-released.md
Comment (SethTisue) [inline blog/_posts/2022-01-17-scala-3.1.1-released.md:36]: This reads strangely. Oh, there's a Scala 3.1.2-RC1 now? But I'm only finding that it in the middle of a paragraph, in the middle of a sentence, in an offhand phrase like "that was just released"?

I'd suggest leading with announcing the availability of 3.1.2-RC1, then go into the material about compatibility.

File: blog/_posts/2022-01-17-scala-3.1.1-released.md
Comment (SethTisue) [inline blog/_posts/2022-01-17-scala-3.1.1-released.md:42]: So what's the upshot for library authors? I'm a library author; what do I do, what do you recommend, what is expected of me, what is the upshot, what do I actually need to know about this right now?

I'd suggest leading with answers to those questions. _Then_ follow it up with technical details and justifications if you feel they are necessary.

Reading this, the situation seems to be rather complex and is still changing rapidly. In that case, it might actually be better to omit the technical details from the release announcement; consider giving people the TL;DR on where things stand, with a link to a contributors thread on Discourse where people can read about and discuss details.

(I might be wrong about what the right way to handle this is, but if I am wrong, that speaks to a lack of clarity in the current text, which should be addressed regardless.)

File: blog/_posts/2022-01-17-scala-3.1.1-released.md
Comment (SethTisue) [inline blog/_posts/2022-01-17-scala-3.1.1-released.md:54]: I doubt this is sufficiently important to repeat in every release announcement.

File: blog/_posts/2022-01-17-scala-3.1.1-released.md
Comment (julienrf) [reply blog/_posts/2022-01-17-scala-3.1.1-released.md:42]:
```suggestion
One important note is that the `-Yscala-release` flag is experimental and not yet available in any stable release of the compiler. It means the build tools, such as SBT or Mill, are not yet aware of its existence. Of course, you can declare usage of the flag in your build file, and the compiler will respect it. However, there may still be some hiccups, like generated artifacts depending on the incorrect version of the stdlib. We will do our best to help fix those minor issues before the stable release of 3.1.2.
```
?

File: blog/_posts/2022-01-17-scala-3.1.1-released.md
Comment (Kordyjan) [reply blog/_posts/2022-01-17-scala-3.1.1-released.md]: Sure, removed.

File: blog/_posts/2022-01-17-scala-3.1.1-released.md
Comment (Kordyjan) [reply blog/_posts/2022-01-17-scala-3.1.1-released.md]: The only link is a pr without any before and after comparison so ¯\_(ツ)_/¯.

File: blog/_posts/2022-01-17-scala-3.1.1-released.md
Comment (Kordyjan) [reply blog/_posts/2022-01-17-scala-3.1.1-released.md]: @bishabosha Does that make sense now?

File: blog/_posts/2022-01-17-scala-3.1.1-released.md
Comment (SethTisue) [reply blog/_posts/2022-01-17-scala-3.1.1-released.md:43]: this is improved. it's a bit hard for me to judge because I already know too much about the situation — I'd suggest running it by a few actual library maintainer or two who haven't been following this stuff closely

File: blog/_posts/2022-01-17-scala-3.1.1-released.md
Comment (Kordyjan) [reply blog/_posts/2022-01-17-scala-3.1.1-released.md]: Yeah, I wanted to write, that now under the hood `<:<` creates implicit conversion inside the body of the function. Also, I wanted to write that now it works the same way as in scala 2, and I'm not sure what else was I thinking, since I finished with writing neither of that. I hope it makes sense now.

File: blog/_posts/2022-01-17-scala-3.1.1-released.md
Comment (SethTisue) [reply blog/_posts/2022-01-17-scala-3.1.1-released.md]: heh. I'd go with "improved" then, or perhaps just omit the bullet entirely, if you aren't confident the changes are truly notable

File: blog/_posts/2022-01-17-scala-3.1.1-released.md
Comment (bishabosha) [inline blog/_posts/2022-01-17-scala-3.1.1-released.md:25]:
```suggestion
- Mirrors can be correctly summoned for hierarchical sum types compiled by Scala 3.0.x. It is now safe to derive a type class for a type dependency that was compiled with a pre-3.1.0 compiler. ([PR #14100](https://github.com/lampepfl/dotty/pull/14100)).
```

File: blog/_posts/2022-01-17-scala-3.1.1-released.md
Comment (tgodzik) [inline blog/_posts/2022-01-17-scala-3.1.1-released.md:43]: > , like generated artifacts depending on the incorrect version of the stdlib

Could you specify exactly why this happens? MAybe `potential problems` section?

```suggestion
One important note is that the `-Yscala-release` flag is experimental and not yet available in any stable release of the compiler. It means the build tools, such as SBT or Mill, are not yet aware of its existence. Of course, you can declare usage of the flag in your build file, and the compiler will respect it. However, there may still be some hiccups, like generated artifacts depending on the incorrect version of the stdlib. We will do our best to help fix those minor issues before the stable release of 3.1.2.
```
?

File: blog/_posts/2022-01-17-scala-3.1.1-released.md
Comment (Kordyjan) [reply blog/_posts/2022-01-17-scala-3.1.1-released.md]: @prolativ Are you going to publish your documentation of `-Yscala-release`?

Comment (prolativ) [reply blog/_posts/2022-01-17-scala-3.1.1-released.md]: Waiting for someone's review
https://github.com/scala/docs.scala-lang/pull/2301/files

File: blog/_posts/2022-02-01-scala-3.1.1-released.md
Comment (prolativ) [inline blog/_posts/2022-02-01-scala-3.1.1-released.md:41]: Actually there are more limitations as described here https://docs.scala-lang.org/scala3/reference/language-versions/binary-compatibility.html
like you cannot depend on libraries compiled for release 3.1 or your code cannot use language features which are not encodable in older TASTy (currently it's not a case but it might be in the future)

Comment (Kordyjan) [reply blog/_posts/2022-02-01-scala-3.1.1-released.md:41]: We don't have any such feature now or planned in the near future. When it happens we will announce that in the respective minor release announcement.

File: blog/_posts/2022-02-01-scala-3.1.1-released.md
Comment (SethTisue) [inline blog/_posts/2022-02-01-scala-3.1.1-released.md:10]: s/is/in/

File: blog/_posts/2022-02-01-scala-3.1.1-released.md
Comment (SethTisue) [inline blog/_posts/2022-02-01-scala-3.1.1-released.md:12]: s/the patch/a patch/

File: blog/_posts/2022-02-01-scala-3.1.1-released.md
Comment (SethTisue) [inline blog/_posts/2022-02-01-scala-3.1.1-released.md:16]: s/the code like below/code like the following/

File: blog/_posts/2022-02-01-scala-3.1.1-released.md
Comment (SethTisue) [inline blog/_posts/2022-02-01-scala-3.1.1-released.md:24]: s/know/learn/ 

(I would not be able to explain why this is more idiomatic, but it is)

File: blog/_posts/2022-02-01-scala-3.1.1-released.md
Comment (SethTisue) [inline blog/_posts/2022-02-01-scala-3.1.1-released.md:34]: s/Together with/Alongside/ — to be clearer that they are simultaneous, yet separate

**Review by SethTisue — COMMENTED** [review]
only nitpicks this round

**Review by prolativ — APPROVED** [review]

**Pull request merged**
